### DISCUSSION TITLE
Link checker error on links to another site page

### DISCUSSION THREAD

**Discussion by clarkWakeland** [alshedivat/al-folio#2607] [Q&A] [answered]
What is the proper way to link a site page from another site page? For example, I have a link on my `about.md` page that looks like 
```
You can read more about my athletic career [here](swimBio).
```

Clicking on this link in my deployed website properly takes me to the `swimBio.md` page, which I want it to do. However, in the link checker workflow, I get this error:

```
### Errors in _pages/about.md

* [ERR] [file:///home/runner/work/clarkwakeland.github.io/clarkwakeland.github.io/_pages/swimBio](file:///home/runner/work/clarkwakeland.github.io/clarkwakeland.github.io/_pages/swimBio) | Failed: Cannot find file

```

I have this same problem for every link to a page on the on site, where it works properly when deployed but fails the link check. What is the proper way to write this link so it doesn't error on the link checker?

Thanks

**Comment by george-gca** [comment, marked as answer]
There are 2 link checkers actions:
- [.github/workflows/broken-links.yml](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/broken-links.yml) - checks for broken links before building the site in almost all files in repo
- [.github/workflows/broken-links-site.yml](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/broken-links-site.yml) - checks for broken links in the built site in offline mode, meaning check for broken links that reference internal files

The correct way to reference pages is:

```markdown
[Link to a page]({% link _pages/projects.md %})

[Link to a project]({% link _projects/1_project.md %})

[Link to a blog post]({% post_url 2015-05-15-images %})
```

Note that you can't add filters to `link` tags. For example, you cannot append a string using Liquid filters, such as `{% link mypage.html | append: "#section1" %}`. To link to sections on a page, you will need to use regular HTML or Markdown linking techniques, as stated in [jekyll docs](https://jekyllrb.com/docs/liquid/tags/).

**Reply by jaiarora0011** [reply]
Is there a way to make sure that references made using these methods open in a new tab?

**Reply by george-gca** [reply]
Try this:
`[link](url){:target="_blank"}`
